handsontable/src/plugins/loading/loading.js

```diff
@@ -0,0 +1,286 @@
+   */
+  #dialogPlugin = null;
+
+  /**
+   * Check if the plugin is enabled in the handsontable settings.
+   *
+   * @returns {boolean}
+   */
+  isEnabled() {
+    return !!this.hot.getSettings()[PLUGIN_KEY];
+  }
+
+  /**
+   * Enable plugin for this Handsontable instance.
+   */
+  enablePlugin() {
+    if (this.enabled) {
+      return;
+    }
+
+    if (this.#dialogPlugin === null) {
+      this.#dialogPlugin = this.hot.getPlugin('dialog');
+
+      if (!this.#dialogPlugin?.isEnabled()) {
+        warn('Dialog plugin is not enabled. Please enable it to use the loading plugin.');
+
+        return;
+      }
+    }
+
+    super.enablePlugin();
+  }
+
+  /**
+   * Update plugin state after Handsontable settings update.
+   */
+  updatePlugin() {
+    this.disablePlugin();
+    this.enablePlugin();
+
+    super.updatePlugin();
+  }
+
+  /**
+   * Disable plugin for this Handsontable instance.
+   */
+  disablePlugin() {
+    this.hide();
+
+    super.disablePlugin();
+  }
+
+  /**
+   * Check if loading dialog is currently visible.
+   *
+   * @returns {boolean}
+   */
+  isVisible() {
+    return this.#dialogPlugin?.isVisible() ?? false;
+  }
+
+  /**
+   * Show loading dialog with optional custom options.
+   *
+   * @param {object} options Custom loading options.
+   * @param {string} options.icon Custom loading icon.
+   * @param {string} options.title Custom loading title.
+   * @param {string} options.description Custom loading description.
+   */
+  show(options = {}) {
+    if (!this.isEnabled() || !this.#dialogPlugin || !this.#dialogPlugin?.isEnabled()) {
+      return;
+    }
+
+    if (this.isVisible()) {
+      this.update(options);
+
+      return;
+    }
+
+    const beforeLoadingShow = this.hot.runHooks('beforeLoadingShow');
+
+    if (beforeLoadingShow === false) {
+      return;
+    }
+
+    this.update(options);
+    this.#dialogPlugin.show();
+
+    this.hot.runHooks('afterLoadingShow');
+  }
+
+  /**
+   * Hide loading dialog.
+   */
+  hide() {
+    if (!this.#dialogPlugin || !this.#dialogPlugin?.isEnabled()) {
+      return;
+    }
+
+    this.#dialogPlugin.hide();
+  }
+
+  /**
+   * Update loading description without hiding/showing the dialog.
+   *
+   * @param {object} options Custom loading options.
+   * @param {string} options.icon Custom loading icon.
+   * @param {string} options.title Custom loading title.
+   * @param {string} options.description Custom loading description.
+   */
+  update(options) {
+    if (!this.isEnabled() || !this.#dialogPlugin || !this.#dialogPlugin?.isEnabled()) {
+      return;
+    }
+
+    this.updatePluginSettings(options);
+
+    const id = this.hot.guid;
+    const icon = this.getSetting('icon');
+    const title = this.getSetting('title');
+    const description = this.getSetting('description');
+
+    const content = loadingContent({
+      id,
+      icon,
+      title,
+      description,
+      phraseTranslator: (...args) => this.hot.getTranslatedPhrase(...args),
+    });
```

Comment:
I think the `loadingContent` function can be simplified (reduce deps to strings only) by passing prepared strings.  
```suggestion
    const id = this.hot.guid;
    const icon = this.getSetting('icon');
    const title = this.getSetting('title') ?? this.hot.getTranslatedPhrase(C.LOADING_TITLE);
    const description = this.getSetting('description');

    const content = loadingContent({
      id,
      icon,
      title,
      description,
    });
```